MerlinAU.sh

```diff
@@ -2670,18 +2589,28 @@ Please manually update to version $minimum_supported_version or higher to use th
     if [ "$releaseVersionNum" -gt "$currentVersionNum" ]
     then
         ##------------------------------------------##
-        ## Modified by ExtremeFiretop [2024-Jan-28] ##
+        ## Modified by ExtremeFiretop [2024-Feb-17] ##
         ##------------------------------------------##
         # Check for the presence of backupmon.sh script
         if [ -f "/jffs/scripts/backupmon.sh" ]; then
             # Extract version number from backupmon.sh
             BM_VERSION=$(grep "^Version=" /jffs/scripts/backupmon.sh | awk -F'"' '{print $2}')
 
-            # Compare current version with the required version
+            # Adjust version format from 1.46 to 1.4.6 if needed
+            DOT_COUNT=$(echo "$BM_VERSION" | tr -cd '.' | wc -c)
+            if [ "$DOT_COUNT" -eq 0 ]; then
+                # If there's no dot, it's a simple version like "1" (unlikely but let's handle it)
+                BM_VERSION="${BM_VERSION}.0.0"
+            elif [ "$DOT_COUNT" -eq 1 ]; then
+                # For versions like 1.46, insert a dot before the last two digits
+                BM_VERSION=$(echo "$BM_VERSION" | sed 's/\.\([0-9]\)\([0-9]\)/.\1.\2/')
```

Comment:
I don't understand why you're converting the "2-dot" version string like "**1.46**" to "**1.4.6**" because the much simpler & accurate method is to convert it to "**1.46.0**" instead, which is the equivalent value when using the new version scheme (i.e. "**major.minor**"  --->  "**major.minor.0**").

The following code should do the job for the "2-dot" version:
`BM_VERSION="${BM_VERSION}.0"`

My 2 cents.

Reply:
> I don't understand why you're converting the "2-dot" version string like "**1.46**" to "**1.4.6**" because the much simpler & accurate method is to convert it to "**1.46.0**" instead, which is the equivalent value when using the new version scheme (i.e. "**major.minor**" ---> "**major.minor.0**").
> 
> The following code should do the job for the "2-dot" version: `BM_VERSION="${BM_VERSION}.0"`
> 
> My 2 cents.
>
Ah, I see why now. BackupMon version went from "1.46" to  "1.5.0"  which is very strange and not conforming to well-established version scheme standards. IOW, "1.46" really means "1.46.0" so the next version should have been "1.47.0" or "1.50.0" In any case, I understand the need to adjust even if it was made due to an error on Viktor's side.

Reply:
Hey @Martinski4GitHub 

Sorry it's been a crazy day! I'm here now though :)

Yes, exactly. The standard changed from 1.46 to .1.5.3 and when converted through our _ScriptVersionStrToNum_ function it would come out looking like: 1046000 instead of 1.4.6 which came out looking like: 1004006

When compared to: 1.5.3 and it came out like: 1005003 so if I didn't add the new not, it would compare : 1046000 vs 1005003 

1046000 is bigger than:
1005003 

and so it would never think the new standard was bigger (1.5.3)
However if I do what I currently do, and format it the same, then it properly compares as: 

1004006 vs
1005003 

And so it properly considers 1.5.3 as larger than 1.46

I know you already clued into all of this, but I just want it in writing in case we ever need to come back to this PR for some reason :)